### ISSUE TITLE
Extensions defined using DSL are not registered on subsequent tasks

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctor-gradle-plugin#166]
There are cases when extensions that are defined using the AsciidoctorJ Groovy DSL inside the task configuration do not get loaded.

Here are the circumstances:
- You have multiple tasks of type AsciidoctorTask
- You define an extension in more than one of the tasks
- You define at least one classpath entry in the buildscript configuration (aside from the plugin itself)

Under these circumstances, the extensions in subsequent tasks in the task chain do not get loaded or used. The problem seems to be that the AsciidoctorJ Groovy DSL dependency isn't being loaded automatically in these circumstances.

I've prepared a project that demonstrates the issue. If you enable the dependency on AsciidoctorJ PDF, the extension don't run in the later tasks.

https://github.com/mojavelinux/asciidoctor-gradle-example-multiple-tasks

Test using:

```
$ ./gradlew asciidoctor asciidoctorB
```

Try with and within the AsciidoctorJ PDF dependency enabled.

**Comment by mojavelinux** [comment]
Perhaps this is a bug in Gradle itself?

**Comment by mojavelinux** [comment]
Note that a workaround for this issue is to simply declare the dependency on the AsciidoctorJ Groovy DSL in the buildscript configuration. That's how I know this is definitely a problem with the automatic registration...because clearly that fills in a void that isn't being filled.

**Comment by maximmo** [comment]
At Hackergarten Basel we tried to investigate this issue. 
We were able to see that AsciidoctorExtensions (package 'org.asciidoctor.groovydsl' in repository asciidoctorj-groovy-dsl) is called for each Asciidoctor-Task. First the extensions() method is called to add the closure. This seems to work fine for all three tasks. Afterwards registerTo() is invoked to run the closure. Unfortunately the static field REGISTERED_EXTENSIONS is empty for the two last tasks (asciidoctorA and asciidoctorB). We were not able to find out why. But maybe this is a hint for the next looking into this issue.
